src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -105,16 +105,29 @@ public void Dispose()
             {
                 await ProcessReceives();
             }
-            catch (SocketException ex) when (ex.SocketErrorCode == SocketError.ConnectionReset)
+            catch (SocketException ex) when (IsConnectionResetError(ex.SocketErrorCode))
             {
-                error = new ConnectionResetException(ex.Message, ex);
-                _trace.ConnectionReset(ConnectionId);
+                _trace.LogTrace("DoReceive Reset (_aborted? {aborted}) {status}: {ex}", _aborted, (int)ex.SocketErrorCode, ex);
+                // A connection reset can be reported as SocketError.ConnectionAborted on Windows
+                if (!_aborted)
```

Comment:
Merge with when condition when removing logs.

Reply:
We're swallowing the exception if aborted, we don't want it to fall through to the next catch.